### DISCUSSION TITLE
Post-generation scripts as Post-Processor

### DISCUSSION THREAD

**Discussion by aleqsss** [actions/virtual-environments#4078] [Image configuration]
As a complement to #4031, I have an additional question:

Why aren't the [Post-generation scripts](https://github.com/actions/virtual-environments/blob/main/docs/create-image-and-azure-resources.md#post-generation-scripts) run via the [Packer Post-Processors](https://www.packer.io/docs/post-processors#post-processors) (perhaps the [Local Shell Post Processor](https://www.packer.io/docs/post-processors/shell-local#local-shell-post-processor)?) instead of the [current suggested and documented procedures](https://github.com/actions/virtual-environments/blob/main/docs/create-image-and-azure-resources.md#running-scripts)?

Is there something that blocks these script from working, if executed by the post-processor?

**Comment by miketimofeev** [comment]
@aleqsss those scripts should be run on the deployed VM because the user used during the packer image generation doesn't exist after the generation is finished.

**Reply by aleqsss** [reply]
Yes but that's the thing with post-processors, they run only after Packer saves the instance as an image (so after the image generation). That's why I though that the [Local Shell Post Processor](https://www.packer.io/docs/post-processors/shell-local#local-shell-post-processor) might would've worked, since it "executes scripts locally during the post processing stage.".

But perhaps this post-processor use the same user as the image generation process itself, then it wouldn't matter and this would make no difference. I was just trying to see if there were some other way, and post-processors came to my mind. 

Anyway, thank you for your input - sticking with custom script extensions on the Virtual Machines it is.

Best regards

**Reply by miketimofeev** [reply]
correct, the post-processor use the same user as the image generation process itself that's why it's not suitable for our needs
